### DISCUSSION TITLE
How to choose a motion plugin?

### DISCUSSION THREAD

**Discussion by veyxov** [folke/flash.nvim#138] [Q&A] [answered]
Hello. There are many motion plugins: lightspeed, leap, flash, hop and others.
What are the main differences and in what kind of workflows and scenarios each shine. Pros and cons of them in your opinion.

Thank you very much for your amazing plugins 🙏

**Comment by folke** [comment, marked as answer]
Best to just check them all out or at least read their docs and see what works best for your workflow.

Flash is search-based, so you can type as many chars as you want before jumping. 
That's probably the main difference with the others you mentioned

**Comment by tony-sol** [comment]
I've started from hop and now try to move to flash, just in curious.
It's always about your own preferences - one may have more funcs, other is easier to setup and configure, etc.

**Comment by jgregoire** [comment]
I started with Leap/Leap Spooky. Flash does all the same things, plus more (search and treesitter integration).

Leap (out of the box, at least) will only find targets that are currently visible, which I found to be a limitation.

This style of motion is more intuitive to me than other plugins, or standard vim motions. I think newbies would have an easier time adopting vim if this was default behavior.

**Comment by MrOxMasTer** [comment]
While I'm trying to set up flash, I'm not satisfied with it at all. There is no 2 character transition. extra labels, instead of putting them in the place where the match is, on the side.
Even if you put it using before = { 0, 0}, it just doesn't work fine with vscode-multi-cursor

**Reply by folke** [reply]
Oh no

**Reply by jgregoire** [reply]
![image-5](https://github.com/user-attachments/assets/e40f70ac-5d2d-4f54-8ccf-4b0b92f96aff)
